File: insertionSort.cpp
#include <iostream>
#include <fstream>
#include <stdlib.h>
#include <time.h>

using namespace std;

void getFromFile(int* numberList, char* name)
{
   ifstream file;
   int num;
   file.open(name);
   for (int i = 0; file >> num; i++)
   {
      numberList[i] = num;//atoi(num.c_str());
   }
}

void swap(int indexOne, int indexTwo, int *numbers)
{
	int temp = numbers[indexOne];
	numbers[indexOne] = numbers[indexTwo];
	numbers[indexTwo] = temp;
}

void display(int * numbers, int size)
{
	for (int i = 0; i < size; i++)
		cout << numbers[i] << " ";
	cout << endl;
}

void insertionSort(int *numbers, int size)
{ 
	for(int i = 0; i < size; i++)
	{
		for(int j = i; j > 0 && numbers[j - 1] > numbers[j]; j--)
		{
			swap (j, j - 1, numbers);
		}
	}
	//display(numbers, size);
}

void recursiveInsertionSort(int *numbers, int last, int size)
{
	if(last > 0)
	{ 
		recursiveInsertionSort(numbers, last - 1, size);
		for(int i = last - 1; i > 0 && numbers[i - 1] > numbers[i]; i--)
			swap(i, i - 1, numbers);
	}
	//display(numbers, size);
}

int main(int argc, char* argv[])
{
   char* name = new char[80];
   int size;
   if (argc > 2)
   {
      name = argv[1];
      size = atoi(argv[2]);
   }
   else
   {
      cout << "Filename: ";
      cin >> name;
      cout << "Size: ";
      cin >> size;
   }
	int numbers[size];
	getFromFile(numbers, name);
           clock_t t;
          t = clock();
	insertionSort(numbers, size);
           t = clock() - t;
	cout << endl;
	display(numbers, size);
         cout << endl << (float)t / CLOCKS_PER_SEC << endl;
	//recursiveInsertionSort(numbers2, size, size);
}

File: insertionSort.java
import java.io.*;
import java.util.Scanner;
class insertionSort
{
   
   public static void getFromFile(int[] numbers)
   {
      try{
         Scanner fileScanner = new Scanner(new File("numberLists/50,000.txt"));
         for (int i = 0; fileScanner.hasNextInt(); i++)
         {
            numbers[i] = fileScanner.nextInt();
         }
      }
      catch (IOException e)
      {
         System.out.println("Wrong file name");
      }
   }
   
	public static void swap(int indexOne, int indexTwo, int numbers[])
	{
		int temp = numbers[indexOne];
		numbers[indexOne] = numbers[indexTwo];
		numbers[indexTwo] = temp;
	}

	private static void display(int numbers[])
	{
		for (int i = 0; i < numbers.length; i++)
			System.out.print(numbers[i] + " ");
		System.out.println();

	}

	public static void insertionSort(int numbers[])
	{
		for(int i = 0; i < numbers.length; i++)
		{
//			display(numbers);
			for(int j = i; j > 0 && numbers[j - 1] > numbers[j]; j--)
			{	
				swap(j, j-1, numbers);
			}
		}
	}

	public static void main(String[] args)
	{
		int[] numbers = new int[50000];
      getFromFile(numbers);
      double startTime = System.currentTimeMillis();
      insertionSort(numbers);
      double stopTime = System.currentTimeMillis();
      double elapsedTime = stopTime - startTime;
      display(numbers);
      System.out.println(elapsedTime/1000.0);
	}
}